### PR TITLE
feat: Implement missing Kraftfile attributes

### LINKED ISSUE #1096: `kraft lib rm` removes some Kraftfile elements
### Describe the bug

before running `kraft lib rm musl`:
```
specification: '0.6'

template: https://github.com/unikraft/app-helloworld.git

name: helloworld
unikraft:
  version: staging
libraries:
  musl:
    version: staging
targets:
  - name: test
    plat: qemu
    arch: x86_64
  - qemu/arm64

```

after running `kraft lib rm musl`:
```
spec: '0.6'
name: helloworld
unikraft:
    version: staging
targets:
    - architecture: x86_64
      platform: qemu
    - architecture: arm64
      platform: qemu
```

As you can see the template was removed and also the target name.

This is most probably an internal problem that reflects also here.

### Steps to reproduce

_No response_

### Expected behavior

The Kraftfile is left intact and only the specified library is removed

### Which architectures were you using or does this bug affect?

_No response_

### Which operating system were you using or does this bug affect?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by mdsahiloss**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes
Fixes #1096

Implements missing attributes of Kraftfile to make Kraftfile's attributes readable internally.

### CODE REVIEW

File: unikraft/app/application.go [hunk 1]
```diff
@@ -915,11 +916,16 @@ func (app application) WithTarget(targ target.Target) (Application, error) {
 // MarshalYAML makes application implement yaml.Marshaller
 func (app application) MarshalYAML() (interface{}, error) {
 	ret := map[string]interface{}{
-		"spec":     schema.SchemaVersionLatest,
 		"name":     app.name,
 		"unikraft": app.unikraft,
 	}
 
+	if app.kraftfile.config["specification"] != nil {
+		ret["specification"] = schema.SchemaVersionLatest
```
Comment (craciunoiuc) [inline unikraft/app/application.go]: Shouldn't this be like this?

```suggestion
	if app.kraftfile.config["specification"] != nil {
		ret["specification"] = app.kraftfile.config["specification"]
```

File: unikraft/app/application.go [hunk 2]
```diff
@@ -915,11 +916,16 @@ func (app application) WithTarget(targ target.Target) (Application, error) {
...
+	} else {
+		ret["spec"] = schema.SchemaVersionLatest
```
Comment (craciunoiuc) [inline unikraft/app/application.go:928]:
```suggestion
		ret["specification"] = schema.SchemaVersionLatest
```

File: unikraft/target/target.go [hunk 3]
```diff
@@ -203,8 +203,16 @@ func TargetPlatArchName(target Target) string {
 
 // MarshalYAML makes TargetConfig implement yaml.Marshaller
 func (tc TargetConfig) MarshalYAML() (interface{}, error) {
-	return map[string]interface{}{
+	ret := map[string]interface{}{
 		"architecture": tc.architecture.Name(),
 		"platform":     tc.platform.Name(),
-	}, nil
+	}
+	// TODO: Needs to check if target name actually exist in Kraftfile.
```
Comment (craciunoiuc) [inline unikraft/target/target.go]:
```suggestion
	// TODO(Sahil): Needs to check if target name actually exist in Kraftfile.
```

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
Hey, left you some comments, also, can you change the commit to:

```
feat: Implement missing Kraftfile attributes
```

File: unikraft/app/application.go (hunk 2 above)
Comment (mdsahiloss) [reply unikraft/app/application.go:928]: Updated conditional block to the following to make it more consistent:
```
if app.kraftfile.config["specification"] != nil {
    ret["specification"] = app.kraftfile.config["specification"]
} else if app.kraftfile.config["spec"] != nil {
    ret["spec"] = app.kraftfile.config["spec"]
} else {
    ret["specification"] = schema.SchemaVersionLatest
}
```

File: unikraft/app/application.go (hunk 1 above)
Comment (mdsahiloss) [reply unikraft/app/application.go]: Yes, we can do this 👍🏽

File: unikraft/target/target.go (hunk 3 above)
Comment (mdsahiloss) [reply unikraft/target/target.go]: Done 👍🏽

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request merged**